src/core_functions/scalar/string/to_base.cpp

```diff
@@ -0,0 +1,79 @@
+#include "duckdb/core_functions/scalar/string_functions.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+
+namespace duckdb {
+
+static const char alphabet[] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+static void ToBaseFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto &input = args.data[0];
+	auto &radix = args.data[1];
+	auto count = args.size();
+
+	BinaryExecutor::Execute<int64_t, int32_t, string_t>(input, radix, result, count, [&](int64_t input, int32_t radix) {
+		if (input < 0) {
+			throw InvalidInputException("'to_base' number must be greater than or equal to 0");
+		}
+		if (radix < 2 || radix > 36) {
+			throw InvalidInputException("'to_base' radix must be between 2 and 36");
+		}
+
+		char buf[64];
+		char *end = buf + sizeof(buf);
+		char *ptr = end;
+
+		do {
+			*--ptr = alphabet[input % radix];
+			input /= radix;
+		} while (input > 0);
+		return StringVector::AddString(result, ptr, end - ptr);
+	});
+}
+
+static void ToBaseFunctionWithPadding(DataChunk &args, ExpressionState &state, Vector &result) {
+	auto &input = args.data[0];
+	auto &radix = args.data[1];
+	auto &min_length = args.data[2];
+	auto count = args.size();
+
+	TernaryExecutor::Execute<int64_t, int32_t, int32_t, string_t>(
+	    input, radix, min_length, result, count, [&](int64_t input, int32_t radix, int32_t min_length) {
+		    if (input < 0) {
+			    throw InvalidInputException("'to_base' number must be greater than or equal to 0");
+		    }
+		    if (radix < 2 || radix > 36) {
+			    throw InvalidInputException("'to_base' radix must be between 2 and 36");
+		    }
+		    if (min_length > 64 || min_length < 0) {
+			    throw InvalidInputException("'to_base' min_length must be between 0 and 64");
+		    }
+
+		    char buf[64];
+		    char *end = buf + sizeof(buf);
+		    char *ptr = end;
+		    do {
+			    *--ptr = alphabet[input % radix];
+			    input /= radix;
```

Comment:
Perhaps we can call `HexStrOperator::Operation` or `BinaryStrOperator::Operation` for a radix of 2 and 16 as those are more optimized (they don't use the more expensive division/modulo operators)

Reply:
Not sure the additional branches required are worth it even if predicted? You would also need to convert the input to varchar before you can use those operators. I think it makes more sense to use the dedicated to_binary/to_hex In case you really need the extra performance.

Reply:
Ah sorry, meant `HexIntegralOperator` and `BinaryIntegralOperator`. I suppose that could also be an optimizer rule (where we convert `to_base(x, 2)` into `to_binary(x)`).